crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -176,6 +135,122 @@ impl From<&Account> for AccountInterface {
+    BasicFungibleFaucet,
+    /// Exposes `auth_tx_rpo_falcon512` procedure from the `miden::contracts::auth::basic` module.
+    ///
+    /// Internal value holds the storage offset where the public key for the RpoFalcon512
+    /// authentication scheme is stored.
+    RpoFalcon512(u8),
+    /// Exposes the procedures vector specified by its internal value.
+    Custom(Vec<AccountProcedureInfo>),
+}
+
+impl AccountComponentInterface {
+    /// Creates a set of [AccountComponentInterface] instances, specifying components which were
+    /// used to create an account with the provided procedures array.
+    pub fn from_procedures(procedures: &[AccountProcedureInfo]) -> BTreeSet<Self> {
+        let mut component_interface_set = BTreeSet::new();
+
+        if basic_wallet_library().mast_forest().procedure_digests().all(|proc_digest| {
+            procedures.iter().any(|proc_info| proc_info.mast_root() == &proc_digest)
+        }) {
+            component_interface_set.insert(AccountComponentInterface::BasicWallet);
+        }
+
+        if basic_fungible_faucet_library()
+            .mast_forest()
+            .procedure_digests()
+            .all(|proc_digest| {
+                procedures.iter().any(|proc_info| proc_info.mast_root() == &proc_digest)
+            })
+        {
+            component_interface_set.insert(AccountComponentInterface::BasicFungibleFaucet);
+        }
+
+        let rpo_falcon_procs = rpo_falcon_512_library()
+            .mast_forest()
+            .procedure_digests()
+            .collect::<Vec<Digest>>();
+
+        debug_assert!(rpo_falcon_procs.len() == 1);
+        let rpo_falcon_proc = rpo_falcon_procs[0];
+
+        procedures.iter().for_each(|proc_info| {
+            if proc_info.mast_root() == &rpo_falcon_proc {
+                component_interface_set
+                    .insert(AccountComponentInterface::RpoFalcon512(proc_info.storage_offset()));
+            }
+        });
+
+        let mut custom_interface_procs_map: BTreeMap<u8, Vec<AccountProcedureInfo>> =
+            BTreeMap::new();
+        procedures.iter().for_each(|proc_digest| {
+            if !basic_wallet_library()
+                .mast_forest()
+                .procedure_digests()
+                .any(|wallet_proc_digest| &wallet_proc_digest == proc_digest.mast_root())
+                && !basic_fungible_faucet_library()
+                    .mast_forest()
+                    .procedure_digests()
+                    .any(|faucet_proc_digest| &faucet_proc_digest == proc_digest.mast_root())
+                && (&rpo_falcon_proc != proc_digest.mast_root())
+            {
+                match custom_interface_procs_map.get_mut(&proc_digest.storage_offset()) {
+                    Some(proc_vec) => proc_vec.push(proc_digest.clone()),
+                    None => {
+                        custom_interface_procs_map
+                            .insert(proc_digest.storage_offset(), vec![proc_digest.clone()]);
```

Comment:
I think we could implement `Copy` for `AccountProcedureInfo` since it's just 40 bytes.